### PR TITLE
soft delete logic fixing tests for BigQuery

### PR DESCRIPTION
**Pull request by heavycrystal**

### CODE REVIEW

**Review by serprex — DISMISSED** [review]

File: flow/connectors/bigquery/bigquery.go [hunk 1]
```diff
@@ -452,7 +452,8 @@ func (c *BigQueryConnector) getTableNametoUnchangedCols(flowJobName string, sync
 	// Prepare the query to retrieve distinct tables in that batch
 	query := fmt.Sprintf(`SELECT _peerdb_destination_table_name,
 	array_agg(DISTINCT _peerdb_unchanged_toast_columns) as unchanged_toast_columns FROM %s.%s
-	 WHERE _peerdb_batch_id > %d and _peerdb_batch_id <= %d GROUP BY _peerdb_destination_table_name`,
+	 WHERE _peerdb_batch_id > %d and _peerdb_batch_id <= %d 
+	 AND _peerdb_record_type != 2 GROUP BY _peerdb_destination_table_name`,
```
Comment (serprex) [inline flow/connectors/bigquery/bigquery.go:456]:
```suggestion
	 WHERE _peerdb_batch_id > %d AND _peerdb_batch_id <= %d AND _peerdb_record_type != 2
	 GROUP BY _peerdb_destination_table_name`,
```

File: flow/e2e/bigquery/bigquery_helper.go [hunk 2]
```diff
@@ -452,3 +452,15 @@ func (b *BigQueryTestHelper) CreateTable(tableName string, schema *model.QRecord
 
 	return nil
 }
+
+func (b *BigQueryTestHelper) RunIntQuery(query string) (int, error) {
```
Comment (serprex) [inline flow/e2e/bigquery/bigquery_helper.go:456]:
```suggestion
func (b *BigQueryTestHelper) RunInt64Query(query string) (int64, error) {
```
No need to convert int64 result to int

**Comment by serprex** [comment]
1. Update PR title that this isn't a draft anymore
2. Seems like tests are slower in this PR, can these test cases be made faster or be split up?

**Comment by heavycrystal** [comment]
I think the BQ tests are becoming slower due to metadata table advisory lock contention. This is needed because BQ doesn't allow concurrent updates.

File: flow/connectors/bigquery/bigquery.go [hunk 3]
```diff
@@ -1078,18 +1080,30 @@ func (c *BigQueryConnector) getStagingTableName(flowJobName string) string {
 // we grab a lock on catalog to ensure that only one job is updating
 // bigquery tables at a time.
 // returns a function to release the lock.
-func (c *BigQueryConnector) grabJobsUpdateLock() (func() error, error) {
+func (c *BigQueryConnector) grabJobsUpdateLock(flowJobName string) (func() error, error) {
```
Comment (serprex) [inline flow/connectors/bigquery/bigquery.go:1083]: This should take mirrorName & include mirrorName in hash

**Review by serprex — APPROVED** [review]

File: flow/connectors/bigquery/bigquery.go [hunk 4]
```diff
@@ -452,7 +452,8 @@ func (c *BigQueryConnector) getTableNametoUnchangedCols(flowJobName string, sync
 	// Prepare the query to retrieve distinct tables in that batch
 	query := fmt.Sprintf(`SELECT _peerdb_destination_table_name,
 	array_agg(DISTINCT _peerdb_unchanged_toast_columns) as unchanged_toast_columns FROM %s.%s
-	 WHERE _peerdb_batch_id > %d and _peerdb_batch_id <= %d GROUP BY _peerdb_destination_table_name`,
+	 WHERE _peerdb_batch_id > %d AND _peerdb_batch_id <= %d AND _peerdb_record_type != 2
```
Comment (serprex) [inline flow/connectors/bigquery/bigquery.go:455]: Include in PR description what `_peerdb_record_type != 2` achieves

**Pull request merged**